.github/workflows/build_documentation.yml

```diff
@@ -0,0 +1,46 @@
+name: Build HTML documentation
+
+on:
+  [push, pull_request]
+
+jobs:
+  build:
+    name: Build documentation
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          python-version: '3.8'
+
+      - name: Install documentation dependencies
+        run: |
+          python -mpip install -r doc/requirements.txt
+
+      - name: Install qutip-qip from GitHub
+        run: |
+          python -mpip install -e .[full]
+          # Install in editable mode so it doesn't matter if we import from
+          # inside the installation directory, otherwise we can get some errors
+          # because we're importing from the wrong location.
+          python -c 'import qutip_qip; print("QuTiP QIP Version:      %s" % qutip_qip.__version__)'
+          python -c 'import qutip; qutip.about()'
+
+      - name: Build documentation
+        working-directory: doc
+        run: |
+          make html SPHINXOPTS="-W --keep-going -T"
+          # Above flags are:
+          #   -W : turn warnings into errors
+          #   --keep-going : do not stop after the first error
+          #   -T : display a full traceback if a Python exception occurs
```

Comment:
I think there's a way to add this into `pyproject.toml` so that these flags are _always_ done, even for users.  I didn't know about it at the time I wrote the qutip/qutip workflow, but if it does exist, it might be an idea to add them there.  I think `pytest` also supports that.

If not `pyproject.toml`, then you could also change the Sphinx Makefile to put them in there.  I don't know why I didn't do that with qutip/qutip - I think that's the right way to do it, and I just got tunnel vision on writing the workflow.